includes/Util.php

```diff
@@ -26,15 +26,18 @@ function generateDomainId() {
 	return substr( $domain_id_full, 0, 10 );
 }
 
-function getDomainId(): string {
+function getDomainID(): string {
+	return MediaWikiServices::getInstance()->getConfigFactory()->makeConfig( 'da' )
+		->get( 'DomainID' );
+}
```

Comment:
It sounds very plausible that MediaWiki is indeed not initialized enough to use this code here. What about directly accessing the global var?

Reply:
yes, on the initial installation the updater registers the sql for table creation. at this point the table for the configs does not exist yet. You could register a post update maintenance script, that will only be executed after all the table creation is done:
https://doc.wikimedia.org/mediawiki-core/master/php/classLoggedUpdateMaintenance.html

`$updater->addPostDatabaseUpdateMaintenance( 'MyClass' );`

`class MyClass extends LoadExtensionSchemaUpdates {
    protected function doProcess() {
        // set the config here
        // return true to make it only execute once
        return true;
    }
}`